### DISCUSSION TITLE
Proposito de migração usando Entity Framework

### DISCUSSION THREAD

**Discussion by 4klayton** [balta-io/forum#325] [Tira Dúvidas] [answered]
Curso Fundamentos do EF > Relacionamentos > Aulas 1, 2 e 3

Boa tarde a todos, acho que me perdi na questão do conceito e não consegui enxergar o real proposito de uma migration.

Ficou claro para mim a questao de que o EF consegue, através do mapeamento tanto fluente quanto por tags gerar um BD, mas o que eu não consegui entender é o proposito dessa ação, na vida real, vou ficar fazendo varias migrações de banco? para que isso serve?

Desculpe se parece uma pergunta óbvia, mas é que realmente não consegui enxergar o propósito.

**Comment by 4klayton** [comment, marked as answer]
Acho que entendi o proposito nas aulas posteriores, a palavra "migração" talvez tenha me confundido. Entendia como migração algo como pegar os dados de uma base e jogar para outra, mas nesse caso, pelo o que entendi, migração é o ato de criar e/ou atualizar a estrutura do banco. É isso? Sendo isso, essa é uma tarefa comum no mercado, sempre existe a necessidade desses ajustes?

**Reply by andrebaltieri** [reply]
Issso!!! Realmente o nome pode confundir hahahahah!!!!
